common/api/core-frontend.api.md

```diff
@@ -8822,6 +8829,8 @@ export interface RenderPlan {
     // (undocumented)
     readonly ao?: AmbientOcclusion.Settings;
     // (undocumented)
+    readonly atmosphericScattering?: AtmosphericScattering.Settings;

```

Comment:
@pmconne If the atmosphere settings are moved onto the environment (next to groundplane & skybox settings), does this field need to exist?

I don't see a field for environment settings in RenderPlan, but the atmosphere settings are expected to be on the RenderPlan in [AtmosphericScatteringUniforms.ts](https://github.com/iTwin/itwinjs-core/blob/97e4bb491bf84c16f90bd3ce6cf0f2de86a03222/core/frontend/src/render/webgl/AtmosphericScatteringUniforms.ts#L52). Should we add an environment property here instead, or should the code in AtmosphericScatteringUniforms be retooled?

Reply:
I think there will be a field in `Environment` called `atmosphere` of type `Atmosphere`, which would replace this field.

Reply:
I've got the `atmosphere` field within `Environment` already wired up locally. I'm just not sure how to propagate that through to the `AtmosphericScatteringUniforms` file because there isn't an `Environment` field on `RenderPlan`; it's only on `DisplayStyle3dSettings`.

Do we need to add an `environment` to the `RenderPlan`, or is there a way to read the settings from elsewhere using the `Target` parameter that's in `[AtmosphericScatteringUniforms.update](https://github.com/iTwin/itwinjs-core/blob/97e4bb491bf84c16f90bd3ce6cf0f2de86a03222/core/frontend/src/render/webgl/AtmosphericScatteringUniforms.ts#L48-L52)`?

Reply:
A RenderPlan is created from a Viewport. A Viewport has DisplayStyle which has DisplayStyleSettings which has Environment (if 3d) which has Atmosphere. Set `RenderPlan.atmosphere` from that.